<a href="https://colab.research.google.com/github/ertomz/h4bl-superfund-website/blob/main/H4BL_Demographic_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np

Let's import some data:

In [ ]:
#census data
#here's the metadata:
#https://www2.census.gov/programs-surveys/popest/technical-documentation/file-layouts/2010-2019/cc-est2019-alldata.pdf

In [ ]:
census = pd.read_csv('census2019countyCLEANED.csv', encoding = 'ISO-8859-1')
census.head(300)

YEAR  AGEGRP  STATE  ...  TOT_POP BAC_MALE BAC_FEMALE
0      12       0      1  ...    55869     5514       6270
1      12       0      1  ...   223234    10346      11008
2      12       0      1  ...    24686     6432       5685
3      12       0      1  ...    22394     3010       1897
4      12       0      1  ...    57826      658        618
..    ...     ...    ...  ...      ...      ...        ...
295    12       0      8  ...   168424     3126       2575
296    12       0      8  ...     6324       79         69
297    12       0      8  ...    11267      117         88
298    12       0      8  ...    25638      222        157
299    12       0      8  ...     6824       71         60

[300 rows x 9 columns]

Since the Superfund site data doesn't have 'County', etc. after the county names, we have to do a bit of data engineering here. 

In [ ]:
census['CTYNAME']=census['CTYNAME'].str.replace('County', '')
census['CTYNAME']=census['CTYNAME'].str.replace('Parish', '')
census['CTYNAME']=census['CTYNAME'].str.replace('Census Area', '')
census['CTYNAME']=census['CTYNAME'].str.replace('Burough', '')
census['CTYNAME']=census['CTYNAME'].str.replace('Municipality', '')
census['CTYNAME']=census['CTYNAME'].str.replace('City and Burough', '')
census['CTYNAME']=census['CTYNAME'].str.replace('city', '')

#census['CTYNAME']=census['CTYNAME'].str.split(' ',expand=True)[0:-1].str[:-1]

census['CTYNAME']=census['CTYNAME'].str.strip(' ')
census

YEAR  AGEGRP  STATE  ...  BAC_MALE BAC_FEMALE          CountyState
0       12       0      1  ...      5514       6270     Autauga, Alabama
1       12       0      1  ...     10346      11008     Baldwin, Alabama
2       12       0      1  ...      6432       5685     Barbour, Alabama
3       12       0      1  ...      3010       1897        Bibb, Alabama
4       12       0      1  ...       658        618      Blount, Alabama
...    ...     ...    ...  ...       ...        ...                  ...
3137    12       0     56  ...       481        389  Sweetwater, Wyoming
3138    12       0     56  ...       147        110       Teton, Wyoming
3139    12       0     56  ...       122        111       Uinta, Wyoming
3140    12       0     56  ...        55         38    Washakie, Wyoming
3141    12       0     56  ...        55         44      Weston, Wyoming

[3142 rows x 10 columns]

Ah, much better :)

Now, we make a 'County, State' column so that we'll be able to match superfund sites to their correct census data (some states have counties with the same name!)

In [ ]:
census['CountyState']= census['CTYNAME'].str.cat(census['STNAME'], sep =", ") 
census

YEAR  AGEGRP  STATE  ...  BAC_MALE BAC_FEMALE          CountyState
0       12       0      1  ...      5514       6270     Autauga, Alabama
1       12       0      1  ...     10346      11008     Baldwin, Alabama
2       12       0      1  ...      6432       5685     Barbour, Alabama
3       12       0      1  ...      3010       1897        Bibb, Alabama
4       12       0      1  ...       658        618      Blount, Alabama
...    ...     ...    ...  ...       ...        ...                  ...
3137    12       0     56  ...       481        389  Sweetwater, Wyoming
3138    12       0     56  ...       147        110       Teton, Wyoming
3139    12       0     56  ...       122        111       Uinta, Wyoming
3140    12       0     56  ...        55         38    Washakie, Wyoming
3141    12       0     56  ...        55         44      Weston, Wyoming

[3142 rows x 10 columns]

###Alright, now moving on to the superfund data:

In [ ]:
#superfund data
superfunds = pd.read_csv("superfunds.csv")
superfunds

Site Name  ...             y
0                            Army Creek Landfill  ...  4.815653e+06
1                Delaware Sand & Gravel Landfill  ...  4.815412e+06
2                        Delaware City PVC Plant  ...  4.805978e+06
3                      Harvey & Knott Drum, Inc.  ...  4.804132e+06
4                               New Castle Steel  ...  4.816336e+06
...                                          ...  ...           ...
1705                              Barber Orchard  ...  4.224634e+06
1706                         Cooper Drum Company  ...  4.021684e+06
1707                            Quanta Resources  ...  4.983520e+06
1708          Griggs & Walnut Ground Water Plume  ...  3.804804e+06
1709  Shenandoah Road Ground Water Contamination  ...  5.090255e+06

[1710 rows x 32 columns]

### Let's explore the makeup of these Superfund sites.
Which are active? Completed? TBD?

In [ ]:
active = superfunds[superfunds["Status"] == "NPL Site"].shape
proposed = superfunds[superfunds["Status"] == "Deleted NPL Site"].shape
deleted = superfunds[superfunds["Status"] == "Proposed NPL Site"].shape

(active, proposed, deleted)

((1259, 32), (404, 32), (47, 32))

There are 1259 active sites, 404 deleted sites, and 47 proposed sites at the moment. 

Let's make another one of those 'County, State' columns for our upcoming join:

In [ ]:
superfunds['CountyState']= superfunds['County'].str.cat(superfunds['State'], sep =", ") 
superfunds

Site Name  ...              CountyState
0                            Army Creek Landfill  ...     New Castle, Delaware
1                Delaware Sand & Gravel Landfill  ...     New Castle, Delaware
2                        Delaware City PVC Plant  ...     New Castle, Delaware
3                      Harvey & Knott Drum, Inc.  ...     New Castle, Delaware
4                               New Castle Steel  ...     New Castle, Delaware
...                                          ...  ...                      ...
1705                              Barber Orchard  ...  Haywood, North Carolina
1706                         Cooper Drum Company  ...  Los Angeles, California
1707                            Quanta Resources  ...       Bergen, New Jersey
1708          Griggs & Walnut Ground Water Plume  ...     Dona Ana, New Mexico
1709  Shenandoah Road Ground Water Contamination  ...       Dutchess, New York

[1710 rows x 33 columns]

And now, to make this into a pandas Series, indexing it by the 'County, State' column we so nicely prepared up above.

In [ ]:
superfunds_series = superfunds.set_index('CountyState').squeeze()
superfunds_series 

Site Name  ...             y
CountyState                                                          ...              
New Castle, Delaware                            Army Creek Landfill  ...  4.815653e+06
New Castle, Delaware                Delaware Sand & Gravel Landfill  ...  4.815412e+06
New Castle, Delaware                        Delaware City PVC Plant  ...  4.805978e+06
New Castle, Delaware                      Harvey & Knott Drum, Inc.  ...  4.804132e+06
New Castle, Delaware                               New Castle Steel  ...  4.816336e+06
...                                                             ...  ...           ...
Haywood, North Carolina                              Barber Orchard  ...  4.224634e+06
Los Angeles, California                         Cooper Drum Company  ...  4.021684e+06
Bergen, New Jersey                                 Quanta Resources  ...  4.983520e+06
Dona Ana, New Mexico             Griggs & Walnut Ground Water Plume  ...  3.804804e+06
Dutchess, New York       Shenandoah Road Ground Water Contamination  ...  5.090255e+06

[1710 rows x 32 columns]

### Let's make a column for the % Black Population for each county by adding the population of black & black-mixed males (`BAC_MALE`) and females (`BAC_FEMALE`)and then dividing them by the county's total population.

In [ ]:
census.head(20)

YEAR  AGEGRP  STATE  ...  BAC_MALE BAC_FEMALE         CountyState
0     12       0      1  ...      5514       6270    Autauga, Alabama
1     12       0      1  ...     10346      11008    Baldwin, Alabama
2     12       0      1  ...      6432       5685    Barbour, Alabama
3     12       0      1  ...      3010       1897       Bibb, Alabama
4     12       0      1  ...       658        618     Blount, Alabama
5     12       0      1  ...      3753       3418    Bullock, Alabama
6     12       0      1  ...      3941       4924     Butler, Alabama
7     12       0      1  ...     11846      13637    Calhoun, Alabama
8     12       0      1  ...      6351       7276   Chambers, Alabama
9     12       0      1  ...       642        604   Cherokee, Alabama
10    12       0      1  ...      2438       2508    Chilton, Alabama
11    12       0      1  ...      2399       2906    Choctaw, Alabama
12    12       0      1  ...      4947       5818     Clarke, Alabama
13    12       0      1  ...       949       1046       Clay, Alabama
14    12       0      1  ...       243        282   Cleburne, Alabama
15    12       0      1  ...      4822       5108     Coffee, Alabama
16    12       0      1  ...      4378       5133    Colbert, Alabama
17    12       0      1  ...      2569       3080    Conecuh, Alabama
18    12       0      1  ...      1637       1726      Coosa, Alabama
19    12       0      1  ...      2314       2674  Covington, Alabama

[20 rows x 10 columns]

###Joining the Superfund and County Data

In [ ]:
census["PCNT_BLACK"] = (census["BAC_MALE"] + census["BAC_FEMALE"]) / census["TOT_POP"]

census_series = census.set_index('CountyState').squeeze()

superfund_and_census = census_series.join(superfunds_series)

#remove any counties that aren't in the superfund dataset
#superfund_and_census = superfund_and_census[superfund_and_census['Site Score'].notna()]
superfund_and_census

YEAR  AGEGRP  ...             x             y
CountyState                              ...                            
Abbeville, South Carolina    12       0  ...           NaN           NaN
Acadia, Louisiana            12       0  ... -1.031013e+07  3.535475e+06
Accomack, Virginia           12       0  ...           NaN           NaN
Ada, Idaho                   12       0  ...           NaN           NaN
Adair, Iowa                  12       0  ...           NaN           NaN
...                         ...     ...  ...           ...           ...
Yuma, Arizona                12       0  ... -1.275597e+07  3.849545e+06
Yuma, Colorado               12       0  ...           NaN           NaN
Zapata, Texas                12       0  ...           NaN           NaN
Zavala, Texas                12       0  ...           NaN           NaN
Ziebach, South Dakota        12       0  ...           NaN           NaN

[4087 rows x 42 columns]

Let's check that we're only using 'all age groups' and data from the most recent census work, July 2019. Then, let's focus in on only the columns that will be helpful for us. AND, we'll create a 'Percent Black' column that tells us the Black proportion of the county's population.

In [ ]:
census = census[census['AGEGRP'] == 0]
census = census[census['YEAR'] == 12]

black_census = census.loc[:, ["STATE", "CountyState", "STNAME", "CTYNAME", "TOT_POP", "BAC_MALE", "BAC_FEMALE"]]
black_census["PCNT_BLACK"] = (black_census["BAC_MALE"] + black_census["BAC_FEMALE"]) / black_census["TOT_POP"]
black_census

STATE          CountyState   STNAME  ... BAC_MALE  BAC_FEMALE  PCNT_BLACK
0         1     Autauga, Alabama  Alabama  ...     5514        6270    0.210922
1         1     Baldwin, Alabama  Alabama  ...    10346       11008    0.095657
2         1     Barbour, Alabama  Alabama  ...     6432        5685    0.490845
3         1        Bibb, Alabama  Alabama  ...     3010        1897    0.219121
4         1      Blount, Alabama  Alabama  ...      658         618    0.022066
...     ...                  ...      ...  ...      ...         ...         ...
3137     56  Sweetwater, Wyoming  Wyoming  ...      481         389    0.020546
3138     56       Teton, Wyoming  Wyoming  ...      147         110    0.010953
3139     56       Uinta, Wyoming  Wyoming  ...      122         111    0.011520
3140     56    Washakie, Wyoming  Wyoming  ...       55          38    0.011915
3141     56      Weston, Wyoming  Wyoming  ...       55          44    0.014292

[3142 rows x 8 columns]

In [ ]:
superfund_and_census.to_csv("Census_County_and_Superfund_ALL.csv", encoding='utf-8', index=False)
superfund_and_census

YEAR  AGEGRP  ...             x             y
CountyState                              ...                            
Abbeville, South Carolina    12       0  ...           NaN           NaN
Acadia, Louisiana            12       0  ... -1.031013e+07  3.535475e+06
Accomack, Virginia           12       0  ...           NaN           NaN
Ada, Idaho                   12       0  ...           NaN           NaN
Adair, Iowa                  12       0  ...           NaN           NaN
...                         ...     ...  ...           ...           ...
Yuma, Arizona                12       0  ... -1.275597e+07  3.849545e+06
Yuma, Colorado               12       0  ...           NaN           NaN
Zapata, Texas                12       0  ...           NaN           NaN
Zavala, Texas                12       0  ...           NaN           NaN
Ziebach, South Dakota        12       0  ...           NaN           NaN

[4087 rows x 42 columns]

Let's filter these so that the % Black is above the national average of 13%. We can go ahead and remove the population data too.

source: https://www.indexmundi.com/facts/united-states/quick-facts/all-states/black-population-percentage#map 



In [ ]:
black_census = black_census[black_census["PCNT_BLACK"] > 0.13]
del black_census['BAC_MALE']
del black_census['BAC_FEMALE']
del black_census['TOT_POP']
del black_census['STATE']
black_census

CountyState     STNAME       CTYNAME  PCNT_BLACK
0           Autauga, Alabama    Alabama       Autauga    0.210922
2           Barbour, Alabama    Alabama       Barbour    0.490845
3              Bibb, Alabama    Alabama          Bibb    0.219121
5           Bullock, Alabama    Alabama       Bullock    0.709930
6            Butler, Alabama    Alabama        Butler    0.455831
...                      ...        ...           ...         ...
2950    Waynesboro, Virginia   Virginia    Waynesboro    0.160097
2951  Williamsburg, Virginia   Virginia  Williamsburg    0.175003
2952    Winchester, Virginia   Virginia    Winchester    0.138151
3087    Milwaukee, Wisconsin  Wisconsin     Milwaukee    0.291590
3098       Racine, Wisconsin  Wisconsin        Racine    0.137333

[747 rows x 4 columns]

In [ ]:
black_census.to_csv("Black_Census_County.csv", encoding='utf-8', index=False)

747 of the 3142 counties in this census data have a Black population percentage that is higher than the national average (13%). That is a little under 24%. 

Let's make this into a pandas Series too, so the join will work.

In [ ]:
black_census_series = black_census.set_index('CountyState').squeeze()
black_census_series

STNAME       CTYNAME  PCNT_BLACK
CountyState                                                
Autauga, Alabama          Alabama       Autauga    0.210922
Barbour, Alabama          Alabama       Barbour    0.490845
Bibb, Alabama             Alabama          Bibb    0.219121
Bullock, Alabama          Alabama       Bullock    0.709930
Butler, Alabama           Alabama        Butler    0.455831
...                           ...           ...         ...
Waynesboro, Virginia     Virginia    Waynesboro    0.160097
Williamsburg, Virginia   Virginia  Williamsburg    0.175003
Winchester, Virginia     Virginia    Winchester    0.138151
Milwaukee, Wisconsin    Wisconsin     Milwaukee    0.291590
Racine, Wisconsin       Wisconsin        Racine    0.137333

[747 rows x 3 columns]

We now see that the counties with less than 13% Black & Black-mixed populations are gone. This step removed just shy of 1200 counties.

### Next, let's join the Superfund data and Black Census data by their County.

In [ ]:
superfund_and_blackcensus = black_census_series.join(superfunds_series)

#remove any counties that aren't in the superfund dataset
superfund_and_blackcensus = superfund_and_blackcensus[superfund_and_blackcensus['Site Score'].notna()]
superfund_and_blackcensus

STNAME    CTYNAME  ...             x             y
CountyState                                       ...                            
Acadia, Louisiana           Louisiana     Acadia  ... -1.031013e+07  3.535475e+06
Aiken, South Carolina  South Carolina      Aiken  ... -9.116190e+06  3.961981e+06
Alachua, Florida              Florida    Alachua  ... -9.164161e+06  3.461842e+06
Albany, New York             New York     Albany  ... -8.215843e+06  5.264863e+06
Alexander, Illinois          Illinois  Alexander  ... -9.959031e+06  4.475186e+06
...                               ...        ...  ...           ...           ...
York, South Carolina   South Carolina       York  ... -9.006211e+06  4.143742e+06
York, South Carolina   South Carolina       York  ... -9.016712e+06  4.159275e+06
York, Virginia               Virginia       York  ... -8.511833e+06  4.463807e+06
York, Virginia               Virginia       York  ... -8.525836e+06  4.473428e+06
York, Virginia               Virginia       York  ... -8.527908e+06  4.478868e+06

[587 rows x 35 columns]

In [ ]:
superfund_and_blackcensus.to_csv("SuperFund_and_BlackCensus.csv", encoding='utf-8', index=False)

Here we have all the Superfunds within counties that have greater than 13% black population, all **587** of them (that's ***34%*** of all Superfunds in the database). Let's just clean it up a smidge:

In [ ]:
del superfund_and_blackcensus["STNAME"]
superfund_and_blackcensus

CTYNAME  PCNT_BLACK  ...             x             y
CountyState                                   ...                            
Acadia, Louisiana         Acadia    0.191780  ... -1.031013e+07  3.535475e+06
Aiken, South Carolina      Aiken    0.265216  ... -9.116190e+06  3.961981e+06
Alachua, Florida         Alachua    0.222459  ... -9.164161e+06  3.461842e+06
Albany, New York          Albany    0.158632  ... -8.215843e+06  5.264863e+06
Alexander, Illinois    Alexander    0.332755  ... -9.959031e+06  4.475186e+06
...                          ...         ...  ...           ...           ...
York, South Carolina        York    0.206457  ... -9.006211e+06  4.143742e+06
York, South Carolina        York    0.206457  ... -9.016712e+06  4.159275e+06
York, Virginia              York    0.156781  ... -8.511833e+06  4.463807e+06
York, Virginia              York    0.156781  ... -8.525836e+06  4.473428e+06
York, Virginia              York    0.156781  ... -8.527908e+06  4.478868e+06

[587 rows x 34 columns]

In [ ]:
superfund_and_blackcensus.index.unique()
# there are 215 unique counties with both superfund sites and a black population >13%

Index(['Acadia, Louisiana', 'Aiken, South Carolina', 'Alachua, Florida',
       'Albany, New York', 'Alexander, Illinois', 'Allegheny, Pennsylvania',
       'Allen, Indiana', 'Allendale, South Carolina', 'Anne Arundel, Maryland',
       'Arapahoe, Colorado',
       ...
       'Wayne, Michigan', 'Webster, Louisiana', 'Westchester, New York',
       'Westmoreland, Virginia', 'Will, Illinois', 'Winn, Louisiana',
       'Winnebago, Illinois', 'Winston, Mississippi', 'York, South Carolina',
       'York, Virginia'],
      dtype='object', name='CountyState', length=215)

- Meanwhile, there are 3142 total counties in this census data. So, 215/3142 = about 7% of US Counties have both Superfund sites and an above average Black proportion of their population.

- If 215 counties have both these qualities, then that means 215/747 = 28.7% of all counties with Black Pop. % > 13% have a Superfund site(s).

- Therefore, while only 747/3142 = 23.7% counties have a Black population percentage higher than the national average, they contain 34.7% of all superfunds in the database.

### Pollution Score Statistics
Comparing Site Scores for counties with above-average black pop. percentage and below-average black pop. percentage.

In [ ]:
superfund_and_blackcensus = superfund_and_blackcensus[superfund_and_blackcensus['Site Score'].notna()]
superfund_and_blackcensus[superfund_and_blackcensus['Site Score'] > 0]
superfund_and_blackcensus['Site Score'].mean()

41.476695059625214

The average site score for counties with a greater than average black pop. percentage is 41.48

Now we will look at the average site score in counties with below-average black pop. percentage.

In [ ]:
low_black_census = census.loc[:, ["STATE", "CountyState", "STNAME", "CTYNAME", "TOT_POP", "BAC_MALE", "BAC_FEMALE"]]
low_black_census["PCNT_BLACK"] = (low_black_census["BAC_MALE"] + low_black_census["BAC_FEMALE"]) / low_black_census["TOT_POP"]
low_black_census

low_black_census = low_black_census[low_black_census["PCNT_BLACK"] <= 0.13]
del low_black_census['BAC_MALE']
del low_black_census['BAC_FEMALE']
del low_black_census['TOT_POP']
del low_black_census['STATE']
low_black_census

CountyState   STNAME     CTYNAME  PCNT_BLACK
1        Baldwin, Alabama  Alabama     Baldwin    0.095657
4         Blount, Alabama  Alabama      Blount    0.022066
9       Cherokee, Alabama  Alabama    Cherokee    0.047565
10       Chilton, Alabama  Alabama     Chilton    0.111326
14      Cleburne, Alabama  Alabama    Cleburne    0.035211
...                   ...      ...         ...         ...
3137  Sweetwater, Wyoming  Wyoming  Sweetwater    0.020546
3138       Teton, Wyoming  Wyoming       Teton    0.010953
3139       Uinta, Wyoming  Wyoming       Uinta    0.011520
3140    Washakie, Wyoming  Wyoming    Washakie    0.011915
3141      Weston, Wyoming  Wyoming      Weston    0.014292

[2395 rows x 4 columns]

In [ ]:
low_black_census_series = low_black_census.set_index('CountyState').squeeze()
low_black_census_series

STNAME     CTYNAME  PCNT_BLACK
CountyState                                         
Baldwin, Alabama     Alabama     Baldwin    0.095657
Blount, Alabama      Alabama      Blount    0.022066
Cherokee, Alabama    Alabama    Cherokee    0.047565
Chilton, Alabama     Alabama     Chilton    0.111326
Cleburne, Alabama    Alabama    Cleburne    0.035211
...                      ...         ...         ...
Sweetwater, Wyoming  Wyoming  Sweetwater    0.020546
Teton, Wyoming       Wyoming       Teton    0.010953
Uinta, Wyoming       Wyoming       Uinta    0.011520
Washakie, Wyoming    Wyoming    Washakie    0.011915
Weston, Wyoming      Wyoming      Weston    0.014292

[2395 rows x 3 columns]

In [ ]:
superfund_and_lowblackcensus = low_black_census_series.join(superfunds_series)

#remove any counties that aren't in the superfund dataset
superfund_and_lowblackcensus = superfund_and_lowblackcensus[superfund_and_lowblackcensus['Site Score'].notna()]
superfund_and_lowblackcensus[superfund_and_gt50blackcensus['Site Score'] > 0]
superfund_and_lowblackcensus

NameError: ignored

Now that the df is set up, let's calculate the avg site score:

In [ ]:
superfund_and_lowblackcensus['Site Score'].mean()

42.054500458295195

The average site score for counties with a less than average black pop. percentage is 42.05, a hair above that for counties with a higher than average black pop. percentage. 

One area of external potential research is to evalauate whether site score for predominantly white areas were exagerated as compared to scores for areas with higher black populations.

### Average Time on the NPL List
Exploring the average time that a site has been waiting on the NPL, based off the percentage of black pop. of that county.

In [ ]:
today = pd.to_datetime("today")
today

Timestamp('2021-02-21 19:09:07.817563')

First, for counties with higher than average black population:

In [ ]:
superfund_and_blackcensus['Construction Completion Date'] = superfund_and_blackcensus['Construction Completion Date'].fillna(today)

superfund_and_blackcensus['Proposed Date'] = pd.to_datetime(superfund_and_blackcensus['Proposed Date'])
superfund_and_blackcensus['Construction Completion Date'] = pd.to_datetime(superfund_and_blackcensus['Construction Completion Date'])


superfund_and_blackcensus['date_diff'] = superfund_and_blackcensus['Construction Completion Date'] - superfund_and_blackcensus['Proposed Date']

superfund_and_blackcensus['date_diff'].mean()

Timedelta('5815 days 06:14:49.531145920')

Now, for counties with lower than average black population percentage:

---



In [ ]:

superfund_and_lowblackcensus['Construction Completion Date'] = superfund_and_lowblackcensus['Construction Completion Date'].fillna(today)

superfund_and_lowblackcensus['Proposed Date'] = pd.to_datetime(superfund_and_lowblackcensus['Proposed Date'])
superfund_and_lowblackcensus['Construction Completion Date'] = pd.to_datetime(superfund_and_lowblackcensus['Construction Completion Date'])


superfund_and_lowblackcensus['date_diff'] = superfund_and_lowblackcensus['Construction Completion Date'] - superfund_and_lowblackcensus['Proposed Date']

superfund_and_lowblackcensus['date_diff'].mean()

Timedelta('5901 days 00:15:12.658257152')

These counties have a slightly higher average resident time on the list.

What about for communities  >50% Black population?

In [ ]:
gt50_black_census = census.loc[:, ["STATE", "CountyState", "STNAME", "CTYNAME", "TOT_POP", "BAC_MALE", "BAC_FEMALE"]]
gt50_black_census["PCNT_BLACK"] = (gt50_black_census["BAC_MALE"] + gt50_black_census["BAC_FEMALE"]) / gt50_black_census["TOT_POP"]
gt50_black_census

gt50_black_census = gt50_black_census[gt50_black_census["PCNT_BLACK"] >= 0.50]
del gt50_black_census['BAC_MALE']
del gt50_black_census['BAC_FEMALE']
del gt50_black_census['TOT_POP']
del gt50_black_census['STATE']

gt50_black_census_series = gt50_black_census.set_index('CountyState').squeeze()
gt50_black_census_series

STNAME     CTYNAME  PCNT_BLACK
CountyState                                           
Bullock, Alabama       Alabama     Bullock    0.709930
Dallas, Alabama        Alabama      Dallas    0.712765
Greene, Alabama        Alabama      Greene    0.804833
Hale, Alabama          Alabama        Hale    0.583578
Lowndes, Alabama       Alabama     Lowndes    0.727740
...                        ...         ...         ...
Emporia, Virginia     Virginia     Emporia    0.662364
Franklin, Virginia    Virginia    Franklin    0.588804
Hampton, Virginia     Virginia     Hampton    0.542592
Petersburg, Virginia  Virginia  Petersburg    0.788873
Portsmouth, Virginia  Virginia  Portsmouth    0.565044

[107 rows x 3 columns]

In [ ]:
superfund_and_gt50blackcensus = gt50_black_census_series.join(superfunds_series)

#remove any counties that aren't in the superfund dataset
superfund_and_gt50blackcensus = superfund_and_gt50blackcensus[superfund_and_gt50blackcensus['Site Score'].notna()]
superfund_and_gt50blackcensus[superfund_and_gt50blackcensus['Site Score'] > 0]

STNAME  ...             y
CountyState                                ...              
Allendale, South Carolina  South Carolina  ...  3.887502e+06
Baltimore, Maryland              Maryland  ...  4.764103e+06
Baltimore, Maryland              Maryland  ...  4.765882e+06
Baltimore, Maryland              Maryland  ...  4.760459e+06
Bibb, Georgia                     Georgia  ...  3.865278e+06
Bibb, Georgia                     Georgia  ...  3.865829e+06
Charles, Maryland                Maryland  ...  4.663309e+06
Coahoma, Mississippi          Mississippi  ...  4.053994e+06
Copiah, Mississippi           Mississippi  ...  3.725340e+06
Crittenden, Arkansas             Arkansas  ...  4.180323e+06
Crittenden, Arkansas             Arkansas  ...  4.180805e+06
Dougherty, Georgia                Georgia  ...  3.705328e+06
Dougherty, Georgia                Georgia  ...  3.713187e+06
Dougherty, Georgia                Georgia  ...  3.706453e+06
Gadsden, Florida                  Florida  ...  3.588761e+06
Gadsden, Florida                  Florida  ...  3.579936e+06
Hampton, Virginia                Virginia  ...  4.448868e+06
Montgomery, Alabama               Alabama  ...  3.812781e+06
Orleans, Louisiana              Louisiana  ...  3.502092e+06
Portsmouth, Virginia             Virginia  ...  4.414951e+06
Portsmouth, Virginia             Virginia  ...  4.412170e+06
Portsmouth, Virginia             Virginia  ...  4.412582e+06
Portsmouth, Virginia             Virginia  ...  4.414317e+06
Prince George's, Maryland        Maryland  ...  4.691873e+06
Prince George's, Maryland        Maryland  ...  4.678513e+06
Prince George's, Maryland        Maryland  ...  4.725850e+06
Richland, South Carolina   South Carolina  ...  4.043078e+06
Richland, South Carolina   South Carolina  ...  4.045303e+06
Richmond, Georgia                 Georgia  ...  3.955747e+06
Richmond, Georgia                 Georgia  ...  3.947112e+06
Richmond, Georgia                 Georgia  ...  3.951343e+06
Shelby, Tennessee               Tennessee  ...  4.169534e+06
Shelby, Tennessee               Tennessee  ...  4.169923e+06
Shelby, Tennessee               Tennessee  ...  4.179536e+06
Shelby, Tennessee               Tennessee  ...  4.169771e+06
Shelby, Tennessee               Tennessee  ...  4.176572e+06
Shelby, Tennessee               Tennessee  ...  4.204670e+06

[37 rows x 35 columns]

In [ ]:
superfund_and_gt50blackcensus['Site Score'].mean()

38.28581395348838

Interesting. Communities with greater than 50% Black population have lower scores for their Superfund sites. I would like to believe this is a good thing, but my understand of institutionalized racism leads me to I wonder if these scores are not entirely accurate. Perhaps, less-needy sites are given 